sdk/src/account.rs

```diff
@@ -21,112 +21,455 @@ pub struct Account {
     pub rent_epoch: Epoch,
 }
 
+/// An Account with data that is stored on chain
+/// This will become a new in-memory representation of the 'Account' struct data.
+/// The existing 'Account' structure cannot easily change due to downstream projects.
+/// This struct will shortly rely on something like the AnAccount trait for access to the fields.
+#[derive(Serialize, Deserialize, PartialEq, Eq, Clone, Default, AbiExample)]
+pub struct AccountSharedData {
+    /// lamports in the account
+    pub lamports: u64,
+    /// data held in this account
+    #[serde(with = "serde_bytes")]
+    pub data: Vec<u8>, // will be: Arc<Vec<u8>>,
+    /// the program that owns this account. If executable, the program that loads this account.
+    pub owner: Pubkey,
+    /// this account's data contains a loaded program (and is now read-only)
+    pub executable: bool,
+    /// the epoch at which this account will next owe rent
+    pub rent_epoch: Epoch,
+}
+
+pub fn accounts_equal<T: AnAccount, U: AnAccount>(me: &T, other: &U) -> bool {
```

Comment:
We haven't gone back andd one this retroactively yet, but in general, all public functions in the sdk should be documented

Reply:
> We haven't gone back andd one this retroactively yet, but in general, all public functions in the sdk should be documented

Where should they be documented? In code or somewhere else also?

Reply:
documenting it in code like this:
https://github.com/solana-labs/solana/blob/a7d5645f72265e243c60084651f30d5dc0b3a5cd/sdk/program/src/pubkey.rs#L158
Will translate into Rust docs like this:
https://docs.rs/solana-program/1.5.13/solana_program/pubkey/struct.Pubkey.html#method.create_program_address

Reply:
I documented this function. I didn't see anything else new and pub except traits in this file.